master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,101 @@
+
+For compatibility with Kubernetes, **Calico network policy** enforcement follows the standard convention for Kubernetes pods:
+- If no network policies apply to a pod, then all traffic to/from that pod is allowed.
+- If one or more network policies apply to a pod with type ingress, then only the ingress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+- If one or more network policies apply to a pod with type egress, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**. Only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
+
+#### Best practice: implicit default deny policy
+
+Whether you use a network policy or global network policy, we recommend that you create a "plain vanilla" default deny policy that does not include other deny rules. This best practice ensures that unwanted traffic is denied by default. Note that implicit default deny policy always occurs last; if any other policy allows the traffic, then the deny does not come into effect. The deny is executed only after all other policies are evaluated. 
+
+### How to
+
+**Best practice**
+
+Although you can use any of the following policies to create default deny for Kubernetes pods, we recommend using the Calico global network policy. A Calico global network policy applies to both workloads (VMs and containers) and hosts (computers that run the hypervisor for VMs, or container runtime for containers). Using a Calico global network policy supports a conservative security stance for protecting resources. 
+
+- [Create default deny traffic Calico global network policy, non-namespaced](#create-default-deny-traffic-calico-global-network-policy-non-namespaced)
+- [Create default deny traffic Calico network policy, namespaced](#create-default-deny-traffic-calico-network-policy-namespaced)
+- [Create default deny traffic Kubernetes policy, namespaced](#create-default-deny-traffic-Kubernetes-policy-namespaced)
+
+#### Create default deny traffic Calico global network policy, non-namespaced
+
+In the following example, we specify a default deny **GlobalNetworkPolicy** for ingress and egress traffic for all workloads and hosts by using the **selector: all()**. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: default-deny
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+```
+
+#### Create default deny traffic Calico network policy, namespaced  
+
+In the following example, we specify a default deny **NetworkPolicy** that denies all traffic for workloads in the namespace, **engineering**. It is an implicit default deny policy without rules; it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. 
```

Comment:
> it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. 

I think we say this above in the "Best practice section. I think it might also be confusing since we _don't_ say this in the GNP example (even though it also applies to that example), which might imply that this example is different in some way in this regard.

Reply:
Fixed.